Fix #744 : Highfi : Topic Practice

virajpatel2411 opened the pull request:
## Explanation 

UI Corrected and made pixel perfect as the mockups.
Issue solved #744.

## Mocks:

Portrait Mock: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/f403add1-079a-4202-9975-efe04c076290/TP-Practice-Unchecked-
Landscape Mock: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/0fa47d7c-e10d-4097-8773-0efa665a323f/TP-Practice-

## The screenshots of new UI are:
![1](https://user-images.githubusercontent.com/36664384/76516144-f0fe2380-6480-11ea-8556-4f7e86d4f66e.jpeg)
![2](https://user-images.githubusercontent.com/36664384/76516151-f3607d80-6480-11ea-8e43-252ef440345e.jpeg)

virajpatel2411 committed:
Fix #744 : UI corrected

virajpatel2411 commented:
@rt4914 Can you please review it now?

veena14cs commented:
Please add mock links as well for reference in this PR.

virajpatel2411 commented:
Portrait Mock: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/f403add1-079a-4202-9975-efe04c076290/TP-Practice-Unchecked-
Landscape Mock: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/0fa47d7c-e10d-4097-8773-0efa665a323f/TP-Practice-

veena14cs commented on app/src/main/res/layout-land/topic_practice_footer_view.xml:
```diff
@@ -15,16 +15,15 @@
-    android:paddingEnd="96dp">
+    android:layout_height="wrap_content">
 
     <Button
```
Check the color of Start button its not white in the mock when it is inactive..

veena14cs commented on app/src/main/res/layout-land/topic_practice_header_view.xml:
```diff
@@ -12,12 +12,10 @@
-    android:layout_marginBottom="8dp"
-    android:paddingStart="48dp"
-    android:paddingEnd="48dp">
+    android:layout_marginLeft="96dp"
```
Use marginStart and marginEnd to support RTL here and elsewhere.

virajpatel2411 replied:
Changed

veena14cs commented on app/src/main/res/layout-land/topic_practice_subtopic.xml:
```diff
@@ -16,8 +16,9 @@
-    android:paddingStart="48dp"
-    android:paddingEnd="48dp"
+    android:layout_marginBottom="16dp"
+    android:layout_marginRight="96dp"
```
Recommended to use margin start and margin end.

virajpatel2411 replied:
Done

veena14cs commented on app/src/main/res/layout-land/topic_practice_subtopic.xml:
```diff
@@ -31,17 +32,15 @@
       android:text="@{viewModel.subtopic.title}"
       android:textColor="@color/oppiaPrimaryText"
       android:textSize="16sp"
+      android:drawableLeft="@android:color/transparent"
```
Ditto change it to drawableStart.

virajpatel2411 replied:
Done

veena14cs commented on app/src/main/res/layout/topic_practice_header_view.xml:
```diff
@@ -6,13 +6,15 @@
     android:layout_marginTop="32dp"
-    android:layout_marginEnd="24dp"
-    android:layout_marginBottom="8dp">
+    android:layout_marginRight="24dp"
```
Ditto.

veena14cs commented on app/src/main/res/layout/topic_practice_subtopic.xml:
```diff
@@ -32,12 +35,10 @@
       android:layout_height="wrap_content"
       android:button="@drawable/selector_checkbox"
       android:checked="@{isChecked}"
+      android:drawableLeft="@android:color/transparent"
```
Ditto.

virajpatel2411 replied:
Done

virajpatel2411 committed:
Color Issue and Left/Right issue solved

virajpatel2411 commented:
@veena14cs Have done the required changes

virajpatel2411 commented:
New image of button is as shown
![1](https://user-images.githubusercontent.com/36664384/76520959-e8f6b180-6489-11ea-96c3-541ed96d10dc.jpeg)

nikitamarysolomanpvt commented:
@virajpatel2411 Please click on the Resolved conversation  Button after resolving and commenting.

virajpatel2411 commented:
@nikitamarysolomanpvt  Done. Can you please finally review it and merge if possible?

nikitamarysolomanpvt commented on app/src/main/res/layout-land/topic_practice_subtopic.xml:
```diff
@@ -31,17 +32,15 @@
       android:id="@+id/subtopic_check_box"
```
It seems failing accessibility test.

virajpatel2411 replied:
I am not getting what needs to be done?

veena14cs replied:
Clickable item should be 48dp or larger. In that case we are not supposed to give fixed height, so you can give min height as 48 dp and height as wrap_content. Make changes in both portrait and landscape to pass accessibility test.

virajpatel2411 replied:
Okay

nikitamarysolomanpvt commented:
@mschanteltc accessibility test is failing for the checkbox PTAL.

mschanteltc commented:
> @mschanteltc accessibility test is failing for the checkbox PTAL.

Is this because it does not meet the minimum touch target size? Or any other accessibility issue?

veena14cs commented on app/src/main/res/layout/topic_practice_fragment.xml:
```diff
@@ -12,7 +12,7 @@
     android:clipToPadding="false"
     android:overScrollMode="never"
-    android:paddingBottom="172dp"
+    android:layout_marginTop="12dp"
```
You can remove this margin top. As you are setting margin top 32 dp in topic_practice_header_view.

virajpatel2411 replied:
No in header I am setting 24dp only...So adding 12dp to 24dp total makes 32dp. I have done in  that way.

veena14cs replied:
Ok got it. Make it consistent in both portrait and landscape.

virajpatel2411 replied:
Done same as mockup

veena14cs commented on app/src/main/res/layout/topic_practice_header_view.xml:
```diff
@@ -21,18 +23,20 @@
       android:text="@string/topic_practice_master_these_skills"
       android:textColor="@color/oppiaPrimaryText"
       android:textSize="20sp"
+      android:layout_marginEnd="88dp"
```
You can give margin end as 32 dp for consistency.

virajpatel2411 replied:
Done

veena14cs commented on app/src/main/res/layout/topic_practice_subtopic.xml:
```diff
@@ -32,12 +35,10 @@
       android:layout_height="wrap_content"
       android:button="@drawable/selector_checkbox"
       android:checked="@{isChecked}"
+      android:drawableStart="@android:color/transparent"
```
Give min height as 48dp to pass accessibility test.

virajpatel2411 replied:
But then the margin between the check boxes is getting increased. @veena14cs

virajpatel2411 replied:
I am giving the checkbox height and width of 14dp. Will it be Okay by it?

veena14cs replied:
Don't give fixed width and height. Share screenshot after giving minHeight.

virajpatel2411 replied:
![1](https://user-images.githubusercontent.com/36664384/76743778-282e4680-6799-11ea-9195-a5a250e70a6a.jpeg)
The margins are getting increased

virajpatel2411 replied:
@veena14cs The screenshot when just minHeight of 48dp given and no margin
![1](https://user-images.githubusercontent.com/36664384/76744810-d8507f00-679a-11ea-8806-067a48094080.jpeg)

veena14cs commented:
> > @mschanteltc accessibility test is failing for the checkbox PTAL.
> 
> Is this because it does not meet the minimum touch target size? Or any other accessibility issue?

Yes it is because of minimum touch target size. We need to make it 48dp.

nikitamarysolomanpvt commented:
> > @mschanteltc accessibility test is failing for the checkbox PTAL.
> 
> Is this because it does not meet the minimum touch target size? Or any other accessibility issue?

Yes checkbox size does not meet the minimum touch target size

virajpatel2411 committed:
Test Cases added

virajpatel2411 committed:
Unnecessary Test Cases removed

mschanteltc commented:
I updated the [mock](https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/f403add1-079a-4202-9975-efe04c076290/TP-Practice-Unchecked-/) and increased the checkbox to be 16px in height and width and increase the textbox size (not font) to be at least 22 px in height. These two can be tapped on to check the box. The options are spaced out 26 px apart, so each option has 13px top and bottom padding, meeting the minimum touch target requirement.

virajpatel2411 committed:
UI updated as per mock to pass accessibility test

virajpatel2411 commented:
@veena14cs @rt4914 @nikitamarysolomanpvt @mschanteltc Please review. I have updated as per new mock

virajpatel2411 committed:
Merge branch 'develop' into ui-update

mschanteltc commented:
> @veena14cs @rt4914 @nikitamarysolomanpvt @mschanteltc Please review. I have updated as per new mock

Spacing LGTM! Can you share a screenshot where a Subtopic's textbox takes up two lines?

virajpatel2411 commented:
> > @veena14cs @rt4914 @nikitamarysolomanpvt @mschanteltc Please review. I have updated as per new mock
> 
> Spacing LGTM! Can you share a screenshot where a Subtopic's textbox takes up two lines?

Imk the text corresponding to checkbox is coming dynamically and at present there is no text longer enough to take two lines

nikitamarysolomanpvt commented:
> > > @veena14cs @rt4914 @nikitamarysolomanpvt @mschanteltc Please review. I have updated as per new mock
> > 
> > 
> > Spacing LGTM! Can you share a screenshot where a Subtopic's textbox takes up two lines?
> 
> Imk the text corresponding to checkbox is coming dynamically and at present there is no text longer enough to take two lines

Please share the updated screen shots so that it will be easy for @mschanteltc to help you out.
Iam unassigning me.Please assign me back once its done and accessibility test is passed.

virajpatel2411 commented:
New Screenshots: 
![2](https://user-images.githubusercontent.com/36664384/77032155-a96c2000-69c9-11ea-9dc6-580f40fe4913.jpeg)
![1](https://user-images.githubusercontent.com/36664384/77032156-ab35e380-69c9-11ea-9d01-b0892ba82de1.jpeg)


Please review the changes. @rt4914 @veena14cs @nikitamarysolomanpvt @mschanteltc

veena14cs commented:
@virajpatel2411 can you please mention your branch name. Branch name should always contain underscore and not spaces and must be in small case.

virajpatel2411 commented:
> @virajpatel2411 can you please mention your branch name. Branch name should always contain underscore and not spaces and must be in small case.

Due to conflicts with other PR I have deleted the repo before a day. So now the branch name isn't visible.
Will it get merged?
@veena14cs @rt4914 @nikitamarysolomanpvt

rt4914 commented:
> > @virajpatel2411 can you please mention your branch name. Branch name should always contain underscore and not spaces and must be in small case.
> 
> Due to conflicts with other PR I have deleted the repo before a day. So now the branch name isn't visible.
> Will it get merged?
> @veena14cs @rt4914 @nikitamarysolomanpvt

No actually because even we cannot check this branch because that branch does not exist. Unless and until we are sure that the code we are merging is correct, we cannot merge the code. So I  think in this case you will need to rework.

rt4914 commented:
@virajpatel2411 Closing this PR as the branch no longer exists.

virajpatel2411 commented:
> > > @virajpatel2411 can you please mention your branch name. Branch name should always contain underscore and not spaces and must be in small case.
> > 
> > 
> > Due to conflicts with other PR I have deleted the repo before a day. So now the branch name isn't visible.
> > Will it get merged?
> > @veena14cs @rt4914 @nikitamarysolomanpvt
> 
> No actually because even we cannot check this branch because that branch does not exist. Unless and until we are sure that the code we are merging is correct, we cannot merge the code. So I think in this case you will need to rework.

I was telling you to review from last 10 days. You requested changes, made changes to mock, I did everthing, changed every change you told me to make, still you don't approve and keep the pr untouched. I will do the work again but this was not fair from your side. @rt4914

rt4914 commented:
> > > > @virajpatel2411 can you please mention your branch name. Branch name should always contain underscore and not spaces and must be in small case.
> > > 
> > > 
> > > Due to conflicts with other PR I have deleted the repo before a day. So now the branch name isn't visible.
> > > Will it get merged?
> > > @veena14cs @rt4914 @nikitamarysolomanpvt
> > 
> > 
> > No actually because even we cannot check this branch because that branch does not exist. Unless and until we are sure that the code we are merging is correct, we cannot merge the code. So I think in this case you will need to rework.
> 
> I was telling you to review from last 10 days. You requested changes, made changes to mock, I did everything, changed every change you told me to make, still you don't approve and keep the pr untouched. I will do the work again but this was not fair from your side. @rt4914

@virajpatel2411 
If i am seeing things correctly, @veena14cs and @nikitamarysolomanpvt have been reviewing your PR on a regular basis. Unless and until your PR is not perfect we cannot merge that PR even if takes a month long time. So I actually don't thing that there has been delay from our side. 
Also, patience is one of the most important part of the development process. Unless and until you do not have that actually it can become hard to work. I would suggest you to follow these steps whenever you are creating a PR.

To improvise on your PR quality you can follow these steps:

1. Once your PR is ready, go to `Files Changed`.
2. Check if there are any unnecessary files changes which you have not introduced but are still showing because of android studio, then revert those changes.
3. Now go through each file in `Files Changed` and open that same file in Android Studio and do following steps:
 - `Control + Alt + L` - formatting shortcut.
 - Try to solve all possible warnings given by android studio.
4. Once all these changes are done, run the test cases which you have written in this PR, if any.

Once all this is done, commit again if there are any changes and then assign it to the reviewers.

virajpatel2411 commented:
@rt4914 The thing is from last few days...the code was perfect. I had made all the changes requested. I am patient and not blaming you. I am new to this open source, don't know much so I know it will take time. But even after changing everything, I will be going to have to do everything again so I was telling that if you would have merged earlier after final changes it would be good as I hadn't deleted the repo before two days. Anyways, I will be making the changes and making a new PR. And thanks for the guidelines of improving your PR. @rt4914
